### DISCUSSION TITLE
How to mount in current(same) directory having static file.

### DISCUSSION THREAD

**Discussion by Bryan-u-2** [fastapi/fastapi#6965] [Questions] [answered]
from fastapi import  FastAPI
from fastapi.staticfiles import StaticFiles
from starlette.responses import FileResponse 

app = FastAPI()
app.mount("./", StaticFiles(directory="./"))

@app.get("/")
async def read_index():
    return FileResponse('./index.html') 

:::Error:::
`    assert path == "" or path.startswith("/"), "Routed paths must start with '/'"
AssertionError: Routed paths must start with '/'`

app = FastAPI()
app.mount("/", StaticFiles(directory="/"))

@app.get("/")
async def read_index():
    return FileResponse('/index.html') 

:::Error:::
`Not Found` in Browser

**Comment by includeamin** [comment]
## directory tree
``` 
├── app.py
└── statics
    └── image.jpg

```
## app.py
``` python
from fastapi import FastAPI
from fastapi.responses import FileResponse
import os

statics = f"{os.getcwd()}/statics"
app = FastAPI()


@app.get("/download/{file_name}")
async def main(file_name: str):
    return FileResponse(os.path.join(statics, file_name))

```

### Note: please follow the [Issue instruction](https://github.com/tiangolo/fastapi/issues/new/choose).

**Comment by Bryan-u-2** [comment]
> you can also just like below example.
> 
> ```
> from fastapi import FastAPI
> from starlette.responses import FileResponse
> import os
> from fastapi import Request
> 
> app = FastAPI()
> 
> @app.get("/get")
> async def read_index():
> 	print(os.getcwd())
> 	return FileResponse('filename.html')
> ```
> 
> **no need of any file path.. If you want the file with the current working directory, you just need to return filename in FileResponse.**

It says (its not working)::
INFO:     127.0.0.1:59686 - "GET /get HTTP/1.1" 200 OK
INFO:     127.0.0.1:59686 - "GET /main.dart.js HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:59686 - "GET /icons/Icon-192.png HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:59686 - "GET /favicon.png HTTP/1.1" 404 Not Found

**Comment by krishnardt** [comment]
Question should have been asked more clearly..
Instead of asking current(same) --> should have been asked "same".

**Comment by Bryan-u-2** [comment, marked as answer]
> > > you can also just like below example.
> > > ```
> > > from fastapi import FastAPI
> > > from starlette.responses import FileResponse
> > > import os
> > > from fastapi import Request
> > > 
> > > app = FastAPI()
> > > 
> > > @app.get("/get")
> > > async def read_index():
> > > 	print(os.getcwd())
> > > 	return FileResponse('filename.html')
> > > ```
> > > 
> > > 
> > > **no need of any file path.. If you want the file with the current working directory, you just need to return filename in FileResponse.**
> > 
> > 
> > It says (its not working)::
> > INFO: 127.0.0.1:59686 - "GET /get HTTP/1.1" 200 OK
> > INFO: 127.0.0.1:59686 - "GET /main.dart.js HTTP/1.1" 404 Not Found
> > INFO: 127.0.0.1:59686 - "GET /icons/Icon-192.png HTTP/1.1" 404 Not Found
> > INFO: 127.0.0.1:59686 - "GET /favicon.png HTTP/1.1" 404 Not Found
> 
> Can I see your code and how you got such responses?

![image](https://user-images.githubusercontent.com/66239183/104990412-858ec600-5a42-11eb-9567-0c0f20b8d618.png)



:::main.py:::
`from fastapi import FastAPI
from starlette.responses import FileResponse
import os
from fastapi import Request

app = FastAPI()

@app.get("/")
async def read_index():
	return FileResponse('index.html')`




::index.html::
![image](https://user-images.githubusercontent.com/66239183/104991052-e1a61a00-5a43-11eb-800b-9df0f5b0de27.png)

**Comment by Bryan-u-2** [comment]
solved thanks for help

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @Bryan-u-2  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
